### PR TITLE
fixes bug where plasmamen don't cool down due to accidental feature loss:

### LINKED ISSUE #14035: Plasmamen don't cool down
(WEB REPORT BY: zealotron REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)
610292f026f13396fe13bf0f105a4920a89695fc 

>General description of the issue
They don't naturally cool down anymore, meaning if they catch on fire once and don't put it out quick enough, they'll burn to death regardless of their suit being on and the fire being out.

>What you expected to happen
to, you know, go down in temperature

>What actually happened
it doesn't go down

>Steps to reproduce if possible
be plasmaman
set yourself on fire
watch as you never cool down

### LINKED ISSUE #17307: Plasmamen violate the laws of physics and retain body heat forever
#### Description of issue
Plasmamen no longer have Celt's thermal regulation

#### Difference between expected and actual behavior
Expected: Plasmamen don't stay hot forever after putting out a fire with their suit

Actual: Plasmamen stay warm forever, possibly presenting themselves as a source of fuel
#### Steps to reproduce 
1.Take off suit
2.Catch fire
3.Put suit back on
#### Specific information for locating

vgstation13/code/modules/mob/living/carbon/human/plasmaman/species.dm
Thermal regulation introduced in:
https://github.com/vgstation-coders/vgstation13/pull/7863
Thermal regulation removed in:
https://github.com/vgstation-coders/vgstation13/pull/8203
( Look for:
if(!H.on_fire) // then the plasma can regulate him.
 			if(H.bodytemperature < T0C+37)
 				H.bodytemperature = min(H.bodytemperature+0.5,T0C+37)
 			else
 				H.bodytemperature = min(H.bodytemperature-0.5,T0C+37)
)

#### Length of time in which bug has been known to occur

Feb 10, 2016

#### Server revision

Game: ce0f1a803fb0dcc8cb7afdd184d73a2fd9e89d1c 
Patches: 78ac2f909b50159df885e55ab563750b9b00c5dd 

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [ ] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)
Issue also stated here https://github.com/vgstation-coders/vgstation13/issues/14035 but I've found where the issue came from so I'm making a new one.

### PR DESCRIPTION
**Pull request by ghost**
this feature was accidentally removed a long time ago in a galaxy far far away

Thermal regulation introduced in:
#7863
Thermal regulation removed in:
#8203 (not deliberately)

fixes #14035
fixes #17307

:cl:
* bugfix: fixes bug where plasmamen don't cool down due to accidental feature loss

### CODE REVIEW

**Comment by ghost** [comment]
HECK

**Comment by ghost** [comment]
I don't know why I wrote unreported bug when I linked (two) issues that reported it

**Comment by Blithering** [comment]
Does this only affect plasma men? At present being on fire is really janky for all races. You take no burn damage but slowly start heating, which carries on after the fire goes out, meaning you can stroll untouched through a blaze, put yourself out, then boil to death sitting in a lukewarm medbay

**Comment by Pathid** [comment]
@Blithering This is specific to plasmamen, yes. Have you tried stripping? (I'm serious, by the way.)

File: code/modules/mob/living/carbon/human/life/handle_breath.dm [hunk 1]
```diff
@@ -112,9 +112,12 @@
 					adjust_fire_stacks(0.5)
 					IgniteMob()
 		else
+			var/obj/item/clothing/suit/space/plasmaman/PS=wear_suit
```
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/human/life/handle_breath.dm]: wear_suit may not specifically be /obj/item/clothing/suit/space/plasmaman should it pass to the else statement, be careful of this.

Comment (DrCelt) [reply code/modules/mob/living/carbon/human/life/handle_breath.dm]: uh, the else is for the !on_fire.. why is it showing as not for the on fire?

Comment (ghost) [reply code/modules/mob/living/carbon/human/life/handle_breath.dm]: oh wait my apologies; no no this is the already existing code for plasmamen as can be seen further down. As it stands, plasmamen can ONLY wear plasmamen suits or it'll runtime anyway, but yeah fair point I can fix that

**Comment by ghost** [comment]
fixed

**Comment by MadmanMartian** [comment]
' I would like for the whole plasmaman thing to be rewritten, so that rather than istype checks for what they're wearing when the person breathes, it triggered an event which the clothing would hook into. But that would be out of scope.'

Something for somebody to do should they look back here in the future

**Pull request merged**